pkgs/applications/misc/1password-gui/default.nix

```diff
@@ -28,15 +28,19 @@
 , nss
 , pango
 , systemd
+, udev
 , xdg-utils
+, ver ? "8.1.1"
+, hash ? "0y39sfhj9xrgprh01i9apzfkqzm6pdhjc8x59x5p5djjjvxbcwmy"
+, channel ? "stable"
```

Comment:
```suggestion
```
If you want to overwrite it locally just use overwriteAttr.

Reply:
With `overwriteAttr`, I was having to override the version and src file, an I kept ending up with inconsistencies:

```
  _1password-gui = local._1password-gui.override {
    version = "8.1.2-10.BETA";
    hash = "0241kxvwyq2j273msqqcf45lc2d2r6njjng5aq554vxsfrw569b7";
    channel = "beta";
    src = pkgs.fetchurl {
      url = "https://downloads.1password.com/linux/tar/${channel}/x86_64/1password-${version}.x64.tar.gz";
      hash = "sha256:${hash}";
    };
  });
```

If I leave out `src` in my override, it doesn't get the replacement hash or version.

Reply:
By contrast, if I make `ver`, `hash` and `channel` into parameters, I can override the derivation like this:

```
  _1password-gui = local._1password-gui.override {
    ver = "8.1.2-10.BETA";
    hash = "0241kxvwyq2j273msqqcf45lc2d2r6njjng5aq554vxsfrw569b7";
    channel = "beta";
  };
```

Reply:
> If I leave out src in my override, it doesn't get the replacement hash or version.

Yeah, because it is fixed output and the hash is not invalidated when the version changes.

> By contrast, if I make ver, hash and channel into parameters, I can override the derivation like this:

Please don't add that to packages where it is not really needed which is not the case here.

Reply:
We release 1Password every week, so I have to do the override every week.

Reply:
I guess, why is this not an okay feature for a derivation?

Reply:
> We release 1Password every week, so I have to do the override every week.

Thats actually pretty easy. Just replace the version in your overlay, invalidate the hash, rebuild and then update the hash.

> I guess, why is this not an okay feature for a derivation?

Because it is nowhere used in nixpkgs. We can't accumulate features for everyone that are unused normally.

Maybe adding an update script would solve the problem. Then updating would be super easy.